tmail-backend/plugins/reporting/src/main/java/com/linagora/tmail/cassandra/CassandraMailReportGenerator.java

```diff
@@ -0,0 +1,142 @@
+package com.linagora.tmail.cassandra;
+
+import static com.datastax.oss.driver.api.core.type.DataTypes.TEXT;
+import static com.datastax.oss.driver.api.core.type.DataTypes.TIMESTAMP;
+import static com.datastax.oss.driver.api.core.type.DataTypes.TIMEUUID;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.selectFrom;
+import static com.linagora.tmail.cassandra.MailReportGeneratorUtils.YEAR_MONTH_FORMATTER;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+
+import jakarta.inject.Inject;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.init.CassandraTableManager;
+import org.apache.james.backends.cassandra.init.CassandraTypesProvider;
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.MaybeSender;
+import org.apache.james.util.DurationParser;
+import org.apache.james.utils.UserDefinedStartable;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.uuid.Uuids;
+import com.github.fge.lambdas.Throwing;
+import com.linagora.tmail.api.MailReportEntry;
+import com.linagora.tmail.api.MailReportGenerator;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class CassandraMailReportGenerator implements MailReportGenerator, UserDefinedStartable {
+    public static final CqlIdentifier ID = CqlIdentifier.fromCql("id");
+    public static final String TABLE_NAME = "mu_mail_report";
+    public static final CqlIdentifier KIND = CqlIdentifier.fromCql("kind");
+    public static final CqlIdentifier SENDER = CqlIdentifier.fromCql("sender");
+    public static final CqlIdentifier RECIPIENT = CqlIdentifier.fromCql("recipient");
+    public static final CqlIdentifier SUBJECT = CqlIdentifier.fromCql("subject");
+    public static final CqlIdentifier DATE = CqlIdentifier.fromCql("date");
+    public static final CqlIdentifier DATE_YEAR_MONTH = CqlIdentifier.fromCql("date_year_month");
+    public static final int TTL = (int) DurationParser.parse(System.getProperty("mu.report.retention", "365d")).getSeconds();
+    public static final CassandraModule MODULE = CassandraModule.builder()
+        .table(TABLE_NAME)
+        .comment("MU sent and received mail audit trail")
+        .options(options -> options.withGcGraceSeconds(0))
+        .statement(statement -> types -> statement
+            .withPartitionKey(DATE_YEAR_MONTH, TEXT)
+            .withClusteringColumn(DATE, TIMESTAMP)
+            .withClusteringColumn(ID, TIMEUUID)
```

Comment:
Please keep it STRICTLY the same

Filtering is OK and was validated with Xavier.

With this the time serie is subject to hot spots and do not scale writes BTW.

Also we do not want to migrate MU data